app/models/round.rb

```diff
@@ -150,29 +150,22 @@ def previous_round
     Round.joins(:competition_event).find_by(competition_event: competition_event, number: number - 1)
   end
 
+  def should_consider_previous_round_results?
+    # All linked rounds except the first one in a chain of linked rounds
+    linked_round.present? && linked_round.first_round_in_link.id != id
+  end
+
   def accepted_registrations
     if number == 1
       registrations.accepted
+    elsif should_consider_previous_round_results?
+      previous_round.accepted_registrations
```

Comment:
You successfully fixed _half_ of my previous comment: The check is now Dual-agnostic, but the body of the `if` still looks at strictly the *immediate* predecessor. Using your new `linked_round.first_round_in_link.registrations` should do fine methinks.